### PR TITLE
[WIP] Adds AudioMAE

### LINKED ISSUE #27453: Audio-MAE - ViTMAE for audio
### Model description

This model is is a Self-supervised Vision Transformer that uses patch reconstruction as the spectrogram task. It extends MAE (which is already on HuggingFace) for audio. This model would be a valuable addition as there doesn't seem to be a self-supervised ViT model on HugginFace currently. AST is the closest and uses supervised pre-training. Conceptually, Audio-MAE is also simpler but achieves comparable performance in their paper.

Some differences compared to the standard MAE Model
- During pre-training local and hybrid attention mechanisms can be used.
- During fine-tuning masking is also used which differs to MAE.

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

**Implementation**
https://github.com/facebookresearch/AudioMAE
Created by Po-Yao Huang
@berniebear on Github

**Pre-trained Weights**
Available in the github repo

**Paper (Masked Autoencoders that Listen)**
https://arxiv.org/abs/2207.06405

### PR DESCRIPTION
**Pull request by ps4vs**
Fixes #27453 Audio-MAE - ViTMAE for audio

@ArthurZucker  @sanchit-gandhi  @ylacombe

### CODE REVIEW

**Comment by ps4vs** [comment]
Hi @NielsRogge, I'm working on this following your tutorial series on contributing model to HF. 

I'm facing issue with downloading and using audioset data. 

Would you mind answering few questions over discord (username:- sivav)

**Comment by NielsRogge** [comment]
Hi @ps4vs sure :)

**Comment by pennychong94** [comment]
hi @ps4vs @ArthurZucker @sanchit-gandhi @ylacombe I am interested in this model and is currently working on finishing up the code. I'm looking to PR in a week or two. Can we work together on this?

**Comment by pennychong94** [comment]
@ps4vs will do
